### PR TITLE
vault: improve error handling and log output

### PR DESCRIPTION
**Pull request by fho**
- log which vault keys are looked up
- ensure exceptions are wrapped as AnsibleError
- use to_native() when wrapping exception messages

Reference: https://docs.ansible.com/ansible/latest/dev_guide/developing_plugins.html#raising-errors

### CODE REVIEW

File: lookup/vault.py [hunk 1]
```diff
@@ -30,108 +33,116 @@
...
+            # Ansible variables are passed via "variables" in ansible 2.x, "inject" in 1.9.x
 
-        basedir = self.get_basedir(variables)
+            basedir = self.get_basedir(variables)
 
-        if hasattr(ansible.utils, 'listify_lookup_plugin_terms'):
-            # ansible-1.9.x
-            terms = ansible.utils.listify_lookup_plugin_terms(terms, basedir, inject)
+            if hasattr(ansible.utils, 'listify_lookup_plugin_terms'):
+                # ansible-1.9.x
```
Comment (nocive) [inline lookup/vault.py:48]: you can discard the support for ansible < 2

File: lookup/vault.py [hunk 2]
```diff
@@ -174,6 +185,8 @@ def _fetch_retry(self, cafile, capath, data, key, vault_token, url, cahostverify
                     raise
 
     def _fetch(self, cafile, capath, data, key, vault_token, url, cahostverify, skipverify):
+        display.vv("looking up vault entry: %s" % (key,))
```
Comment (nocive) [inline lookup/vault.py:188]: python 3 favors the new `.format()` usage in case you wanna use that instead.

Comment (fho) [reply lookup/vault.py:188]: I'll keep it as it is, it works :-)

In python3.6 they introduced fstrings for formatting: https://docs.python.org/3/reference/lexical_analysis.html#f-strings

File: lookup/vault.py (hunk 1 above)
Comment (fho) [reply lookup/vault.py:48]: Lets do that in a separate PR,  there are more ansible <2 artifacts in `run()` which are not so straightforward to remove and will need to be tested again

**Comment by nocive** [comment]
merge then?

**Pull request merged**